app/locations/none.js

```diff
@@ -0,0 +1,37 @@
+import Ember from 'ember';
+
+const {
+  computed: { reads },
+  inject: { service },
+  get
+} = Ember;
+
+const TEMPORARY_REDIRECT_CODE = 307;
+
+export default Ember.NoneLocation.extend({
+  implementation: 'fastboot',
+  fastboot: service(),
+
+  _response: reads('fastboot.response'),
+  _request: reads('fastboot.request'),
+
+  setURL(path) {
+    const currentPath = get(this, 'path');
+    const isInitialPath = !currentPath || currentPath.length === 0;
+    const isTransitioning = currentPath !== path;
+    let response = get(this, '_response');
+
+    if (isTransitioning && !isInitialPath) {
+      let protocol = get(this, '_request.protocol');
+      let host = get(this, '_request.host');
+      let redirectURL = `${protocol}://${host}${path}`;
+
+      response.statusCode = TEMPORARY_REDIRECT_CODE;
+      response.headers.set('location', redirectURL);
+    }
+
+    response.headers.set('x-fastboot-path', path); // for testing and debugging
+
+    this._super(...arguments);
+  }
+});
```

Comment:
the [hardcoded lines](https://github.com/emberjs/ember.js/blob/v2.5.0/packages/ember-application/lib/system/application-instance.js#L401-L405)